### ISSUE TITLE
Self-unification segmentation fault

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#480]
v1.23.0
```
?- .(X,[X]) = X. 
Segmentation fault (core dumped)
```

**Comment by flexoron** [comment]
```
(gdb) where
#0  0x00007ffff793113b in __vfprintf_internal () from /usr/lib/libc.so.6
#1  0x00007ffff7943f8a in __vsnprintf_internal () from /usr/lib/libc.so.6
#2  0x00007ffff791e396 in snprintf () from /usr/lib/libc.so.6
#3  0x0000555555608ce0 in print_term_to_buf (q=0x55555576bbd0, dst=0x2757 <error: Cannot access memory at address 0x2757>, 
    dstlen=0, c=0x55555576b258, c_ctx=0, running=-1, cons=false, depth=0) at src/print.c:765
#4  0x0000555555609419 in print_term_to_buf (q=0x55555576bbd0, dst=0x2757 <error: Cannot access memory at address 0x2757>, 
    dstlen=0, c=0x55555576b258, c_ctx=0, running=-1, cons=false, depth=1) at src/print.c:783
#5  0x0000555555609419 in print_term_to_buf (q=0x55555576bbd0, dst=0x2756 <error: Cannot access memory at address 0x2756>, 
    dstlen=0, c=0x55555576b258, c_ctx=0, running=-1, cons=false, depth=1) at src/print.c:783

 :  address 0x2757
 :  upstairs
 :  address 0x2
 :  address 0x1

#10073 0x0000555555609419 in print_term_to_buf (q=0x55555576bbd0, dst=0x2 <error: Cannot access memory at address 0x2>, dstlen=0, c=0x55555576b258, c_ctx=0, running=-1, cons=false, depth=1) at src/print.c:783
#10074 0x0000555555609419 in print_term_to_buf (q=0x55555576bbd0, dst=0x1 <error: Cannot access memory at address 0x1>, dstlen=0, c=0x55555576b258, c_ctx=0, running=-1, cons=false, depth=1) at src/print.c:783
#10075 0x000055555560e3db in print_term (q=0x55555576bbd0, fp=0x7ffff7a87520 <_IO_2_1_stdout_>, c=0x55555576b258, c_ctx=0, running=-1) at src/print.c:1280
#10076 0x0000555555615e03 in dump_vars (q=0x55555576bbd0, partial=false) at src/query.c:1510
#10077 0x000055555561732c in start (q=0x55555576bbd0) at src/query.c:1863
```

**Comment by flexoron** [comment]
segfault fixed with v1.23.18
